spec/requests/auth_controller_spec.rb

```diff
@@ -0,0 +1,185 @@
+require 'rails_helper'
+
+# NOTE:
+# System tests are slow and tend to be flaky. Many of these tests used to be
+# in a system test (login_spec.rb) but this request test is faster and more reliable.
+RSpec.describe AuthController, type: :request, queue_adapter: :test do
+
+  # This is the user that gets looked up in the DB:
+  let!(:user) { FactoryBot.create(:admin_user, email: 'the_user@sciencehistory.org', password: "goatgoat") }
+
+  context "using Microsoft SSO to log in" do
+
+    # An authenticated email. This email address belongs to a person who has gotten authenticated.
+    let(:incoming_email) { 'the_user@sciencehistory.org' }
+  
+    before do
+      allow(ScihistDigicoll::Env).to receive(:lookup).and_call_original
+      allow(ScihistDigicoll::Env).to receive(:lookup).with(:log_in_using_microsoft_sso).and_return(true)
+      Rails.application.reload_routes!
```

Comment:
Thsi will end up doing the reload_routes before every example -- is it worth seeing how much this slows down our tests? Maybe not, as it's a pain to do it any way other than this reliably!

Reply:
It's fast enough - roughly 10 ms every time you reload the routes. I did add a comment to that effect.